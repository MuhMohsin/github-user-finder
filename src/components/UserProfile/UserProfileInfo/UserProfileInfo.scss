$badgeColor: #fabd22;
$secondary: #ababab;
$btnBgColor: #1d1d1d;
$color: #2a292e;
$whiteColor: #ffffff;
$borderColor: #dadada;
$red: red;

/** Dark Mode Color */
$dark-color-text: #d1d1d1;

:root {
  --color-yellow: #{$badgeColor};
  --color-text: #{$color};
  --color-border: #{$borderColor};
  --color-button: #{$btnBgColor};
  --color-white: #{$whiteColor};
  --color-bg: #{$btnBgColor};
  --color-light: #{$secondary};
  --color-hover-bg: #{$whiteColor};
  --color-hover-border-color: #{$badgeColor};
  --color-hover-color: #{$badgeColor};
}

.profile.darkmode {
  --color-bg: #363636;
  --color-text: #d1d1d1;
  --color-button: #{$btnBgColor};
  --color-hover-bg: #{$badgeColor};
  --color-hover-border-color: #{$badgeColor};
  --color-hover-color: #363636;
  --color-white: #{$whiteColor};
  --color-light: #9f9f9f;
}

.profile {
  &__image {
    width: 150px;
    height: 150px;
    border-radius: 50%;
    overflow: hidden;
    margin: 11px auto;

    & img {
      object-fit: cover;
    }
  }

  &__badge {
    color: var(--color-yellow);
    position: absolute;
    top: 10px;
    left: 10px;
    font-weight: 800;
  }

  &__name {
    font-size: 18px;
    text-align: center;
    font-weight: 800;
    color: var(--color-text);
  }
  &__skill {
    font-size: 10px;
    display: block;
    color: var(--color-light);
    line-height: 20px;
    font-weight: 500;
  }

  &__action {
    background-color: var(--color-bg);
    border: 1px solid transparent;
    border-radius: 20px;
    width: 120px;
    height: 35px;
    color: var(--color-white);
    display: block;
    cursor: pointer;
    transition: 0.3s;
    margin: 10px auto 20px;

    &:hover {
      background-color: var(--color-hover-bg);
      border-color: var(--color-hover-border-color);
      color: var(--color-hover-color);
    }
  }

  /** Counter */
  &__counter {
    display: flex;
    justify-content: space-evenly;
    text-align: center;
  }

  &__follow__label {
    font-size: 10px;
    color: var(--color-light);
  }

  &__follow__number {
    margin-top: 6px;
    font-size: 20px;
    font-weight: 800;
    color: var(--color-text);
  }
}
