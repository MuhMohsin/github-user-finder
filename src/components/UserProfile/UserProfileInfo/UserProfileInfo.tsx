import * as React from "react"
import "./UserProfileInfo.scss"
import { useSelector } from "react-redux"

interface Props {
  userInfo: any
}

export const UserProfileInfo: React.FC<Props> = ({ userInfo }) => {
  const isDark = useSelector(state => state.isDark)

  return (
    <div className={`profile ${isDark && `darkmode`}`}>
      <div className="profile__badge">PRO</div>
      <div className="profile__image">
        <img src={userInfo.avatarUrl} alt={`${userInfo.name} Image Of GUF`} />
      </div>
      <div className="profile__body">
        <h2 className="profile__name">
          {userInfo.name ? userInfo.name : "No Name"}
          <span className="profile__skill">
            {userInfo.bio
              ? userInfo.bio.length > 20
                ? `${userInfo.bio.slice(0, 20)}...`
                : userInfo.bio
              : "No Bio"}
          </span>
        </h2>
        <button className="profile__action">FOLLOW</button>

        <div className="profile__counter">
          <div className="profile__follow">
            <span className="profile__follow__label">FOLLOWERS</span>
            <p className="profile__follow__number">
              {userInfo.followers ? userInfo.followers.totalCount : "0"}
            </p>
          </div>

          <div className="profile__follow">
            <span className="profile__follow__label">FOLLOWINGS</span>
            <p className="profile__follow__number">
              {userInfo.followers ? userInfo.following.totalCount : "0"}
            </p>
          </div>

          <div className="profile__follow">
            <span className="profile__follow__label">STARS</span>
            <p className="profile__follow__number">
              {userInfo.followers
                ? userInfo.starredRepositories.totalCount
                : "0"}
            </p>
          </div>
        </div>
      </div>
    </div>
  )
}
