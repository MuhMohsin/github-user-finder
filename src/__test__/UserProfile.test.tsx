import { configure } from "enzyme"
import EnzymeAdapter from "enzyme-adapter-react-16"
import configureStore from "redux-mock-store" //ES6 modules
import React from "react"
import { shallow, mount } from "enzyme"
import { UserProfileInfo } from "../components/UserProfile/UserProfileInfo/UserProfileInfo"
import { Provider } from "react-redux"
import action from "../state/createStore"

const middlewares = []
const mockStore = configureStore(middlewares)

configure({ adapter: new EnzymeAdapter() })
// You would import the action from your codebase in a real scenario
describe("SearchBox", () => {
  let state
  let store
  let wrapper
  beforeEach(() => {
    state = {
        userFetchedData: {},
        isLoading: true
    }
    store = mockStore({
      startup: { complete: false },
    })
    wrapper = shallow(
      <Provider store={store}>
        <UserProfileInfo {...state} />
      </Provider>
    )
  })
  test("Should render full", () => {
      console.log(wrapper.debug());
    expect(wrapper).toMatchSnapshot('hello')
  })

})
