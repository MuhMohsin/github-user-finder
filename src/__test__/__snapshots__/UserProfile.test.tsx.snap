// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Should render full 1`] = `ShallowWrapper {}`;

exports[`SearchBox Should render full: hello 1`] = `ShallowWrapper {}`;
